src/main/java/com/google/sps/servlets/DrinkRatingServlet.java

```diff
@@ -0,0 +1,142 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+// Servlet to process users rating a drink from a coffee shop
+@WebServlet("/drink-rating")
+public class DrinkRatingServlet extends HttpServlet {
+
+  private final String maxRating = "/5.0";
+  private List<String> knownDrinks = new ArrayList<>();
+  public Map<String, Integer> votes = createMap();
+
+  private Map<String, Integer> createMap() {
+    Map<String, Integer> map = new LinkedHashMap<>();
+
+    map.put("1", 0);
+    map.put("2", 0);
+    map.put("3", 0);
+    map.put("4", 0);
+    map.put("5", 0);
+
+    return map;
+  }
+
+  private double getAverageRating(Map<String, Integer> votes) {
+
+    int totalRatings = 0;
+    int totalVotes = 0;
+
+    for (int i = 1; i <= votes.size(); i++) {
+      String str = Integer.toString(i);
+      totalRatings = i * votes.get(str);
+      totalVotes += votes.get(str);
+    }
+
+    double average = (double) (totalRatings / totalVotes);
+    average = Math.round(average * 10) / 10.0;
```

Comment:
Is this trying to round `average` to 1 decimal place? (Could add a comment here explaining what this does)

Reply:
Yes that's what it's doing. I'll add that comment.

Reply:
Even better, why not extract this into it's own method called roundToOneDecimalPlace()? Makes the code easier to follow, reduces method size (smaller is better), and results in self-documenting code.

Reply:
Ok I'll make that change.